src/extractor/guidance/intersection_normalizer.cpp

```diff
@@ -21,178 +20,125 @@ IntersectionNormalizer::IntersectionNormalizer(
     const util::NameTable &name_table,
     const SuffixTable &street_name_suffix_table,
     const IntersectionGenerator &intersection_generator)
-    : node_based_graph(node_based_graph), node_coordinates(node_coordinates),
-      name_table(name_table), street_name_suffix_table(street_name_suffix_table),
-      intersection_generator(intersection_generator)
+    : node_based_graph(node_based_graph), intersection_generator(intersection_generator),
+      mergable_road_detector(node_based_graph,
+                             node_coordinates,
+                             intersection_generator,
+                             intersection_generator.GetCoordinateExtractor(),
+                             name_table,
+                             street_name_suffix_table)
 {
 }
 
 std::pair<IntersectionShape, std::vector<std::pair<EdgeID, EdgeID>>> IntersectionNormalizer::
 operator()(const NodeID node_at_intersection, IntersectionShape intersection) const
 {
+    const auto intersection_copy = intersection;
     auto merged_shape_and_merges =
         MergeSegregatedRoads(node_at_intersection, std::move(intersection));
+    if (!merged_shape_and_merges.second.empty())
```

Comment:
This is what I mean with the pairs above: what is .second here? What is .first below?